### PR TITLE
Use SecurityTokenDescriptor when creating tokens

### LINKED ISSUE #2993: Adding new parameters to APIs for creating tokens results in parameter explosion
Any time a new parameter needs to be added to configure token creation various APIs need to be updated. These APIs have a lot of parameters already, and the `SecurityTokenDescriptor` class already exists to avoid this. Update all APIs so that internally they're using this class so that eventually the APIs with lots of parameters can be deprecated and removed.

### PR DESCRIPTION
**Pull request by msbw2**
Use `SecurityTokenDescriptor` when creating tokens.

### CODE REVIEW

**Comment by keegan-caruso** [comment]
Let's make sure there is an issue so we can track and include in release notes easier.

**Comment by msbw2** [comment]
> Let's make sure there is an issue so we can track and include in release notes easier.

@keegan-caruso Updated.

**Comment by armandRobled** [comment]
"Suscribe"

Luis Armando Robledo salazar
________________________________

File: src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs [hunk 1]
```diff
@@ -35,14 +35,7 @@ public virtual string CreateToken(string payload)
...
 
-            return CreateToken(
-                    payload,
-                    null,
-                    null,
-                    null,
-                    null,
-                    null,
-                    null);
+            return CreateToken(payload, new SecurityTokenDescriptor());
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs]: SecurityTokenDescriptor can be a static instance in this case to save the allocation.

**Review by keegan-caruso — APPROVED** [review]
LGTM, just that one alloc we can avoid.

**Review by westin-m — APPROVED** [review]

**Pull request merged**